python_modules/libraries/dagster-shell/dagster_shell/utils.py

```diff
@@ -99,7 +99,7 @@ def pre_exec():
             stderr=stderr_pipe,
             cwd=cwd,
             env=env,
-            preexec_fn=pre_exec,
+            preexec_fn=pre_exec,  # noqa: PLW1509
```

Comment:
fascinating https://discuss.python.org/t/why-preexec-fn-in-subprocess-popen-may-lead-to-deadlock/16908/3